v1/StorageSpec.dhall

```diff
@@ -4,10 +4,39 @@ let Kubernetes = imports.Kubernetes
 
 let EmbeddedPersistentVolumeClaim = ./EmbeddedPersistentVolumeClaim.dhall
 
+let Common = { disableMountSubPath : Optional Bool }
+
+let EmptyDir = Common ⩓ { emptyDir : Kubernetes.EmptyDirVolumeSource.Type }
+
+let VolumeClaimTemplate =
+      Common ⩓ { volumeClaimTemplate : EmbeddedPersistentVolumeClaim.Type }
+
+let common = { disableMountSubPath = None Bool }
+
 let StorageSpec =
-      < EmptyDir : { emptyDir : Kubernetes.EmptyDirVolumeSource.Type }
-      | VolumeClaimTemplate :
-          { volumeClaimTemplate : EmbeddedPersistentVolumeClaim.Type }
-      >
+      { Type =
+          < EmptyDir : EmptyDir | VolumeClaimTemplate : VolumeClaimTemplate >
+      , EmptyDir = { Type = EmptyDir, default = common }
+      , VolumeClaimTemplate = { Type = VolumeClaimTemplate, default = common }
+      }
```

Comment:
```dhall
let Options = { EmptyDir , VolumeClaimTemplate }
let StorageSpec = 
          < EmptyDir : Options.EmptyDir | VolumeClaimTemplate : Options.VolumeClaimTemplate >
in    { Type = StorageSpec
      , EmptyDir = \(empty-dir : Options.EmptyDir.Type) -> StorageSpec.EmptyDir empty-dir
      , VolumeClaimTemplate = \(volume-claim-template : Options.VolumeClaimTemplate) -> StorageSpec.VolumeClaimTemplate volume-claim-template
      , Options
      }
```

Reply:
I am not in favor of that, declining it.